### PR TITLE
Test collecting lsp4mp extensions

### PR DESCRIPTION
**Pull request by rzgry**
Adds test for collecting lsp4mp extensions and java version requirements. 

Switches mocha interface from `tdd` to `bdd` so we don't need to import `describe` / `it` in test files.

Also removes the `vscodeTests` directory since vscode-microprofile has no `vscodeUITests` folder like vscode-quarkus so the extra `vscodeTests` folder is not needed.

Signed-off-by: Ryan Zegray <ryan.zegray@ibm.com>

### CODE REVIEW

File: src/test/suite/languageServer/plugin.test.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import * as fs from "fs";
```
Comment (xorye) [inline src/test/suite/languageServer/plugin.test.ts:1]: @fbricon do we need a special header here?

**Comment by rzgry** [comment]
FYI if you merge https://github.com/redhat-developer/vscode-microprofile/pull/8 first. I can rebase and update the tests to include document selectors as well.

File: .vscode/launch.json [hunk 2]
```diff
@@ -41,10 +41,10 @@
...
 			"args": [
 				"--extensionDevelopmentPath=${workspaceFolder}",
-				"--extensionTestsPath=${workspaceFolder}/out/test/vscodeTest/suite/index"
+				"--extensionTestsPath=${workspaceFolder}/out/test/suite/index"
 			],
 			"outFiles": [
-				"${workspaceFolder}/out/test/vscodeTest/**/*.js"
+				"${workspaceFolder}/out/test/**/*.js"
 			],
 			"preLaunchTask": "npm: watch"
```
Comment (xorye) [inline .vscode/launch.json:49]: Not a problem with this PR but, can we change this to `"preLaunchTask": "npm: pretest"` instead?

**Comment by xorye** [comment]
Thanks @rzgry!

**Pull request merged**